spec/pagseguro/payment_request/request_serializer_spec.rb

```diff
@@ -0,0 +1,251 @@
+require "spec_helper"
+
+describe PagSeguro::PaymentRequest::RequestSerializer do
+  context '#to_xml_params' do
+    subject { described_class.new(payment) }
+
+    let(:payment) { PagSeguro::PaymentRequest.new }
+
+    it 'should serializer primary receivers' do
+      payment.primary_receiver = 'a@example.com'
+      expect(subject.to_xml_params).to match /<primaryReceiver>.*<email>a@example.com<\/email>.*<\/primaryReceiver>/m
+    end
+
+    it 'should serializer other receivers' do
+      payment.receivers = [
+        {
+          email: 'a@example.com',
+          split: {
+            amount: 10,
+            fee_percent: 11,
+            rate_percent: 12
+          }
+        }
+      ]
+      expect(subject.to_xml_params).to match /<receivers>.*<receiver>.*<email>a@example.com<\/email>/m
+    end
+
+    it 'should serializer sender' do
+      payment.sender = PagSeguro::Sender.new(
+        name: 'Sender',
+        email: 'b@example.com',
+        phone: PagSeguro::Phone.new(area_code: 51, number: 2)
+      )
+      expect(subject.to_xml_params).to match /<sender>.*<name>Sender<\/name>.*<email>b@example.com<\/email>.*<phone>.*<areaCode>51<\/areaCode>.*<number>2<\/number>/m
+    end
+
+    it 'should serializer currency' do
+      payment.currency = 'BRL'
+      expect(subject.to_xml_params).to match %r[<currency>BRL</currency>]
+    end
+
+    it 'should serializer reference' do
+      payment.reference = 'order123'
+      expect(subject.to_xml_params).to match %r[<reference>order123</reference>]
+    end
+
+    it 'should serializer redirect url' do
+      payment.redirect_url = 'http://example.com'
+      expect(subject.to_xml_params).to match %r[<redirectURL>http://example.com</redirectURL>]
+    end
+
+    it 'should serializer notification url' do
+      payment.notification_url = 'http://example.com'
+      expect(subject.to_xml_params).to match %r[<notificationURL>http://example.com</notificationURL>]
+    end
+
+    context 'should serializer item' do
+      before do
+        payment.items << { id: 1, description: 'Description#1', quantity: 2, amount: 3, weight: 4, shipping_cost: 5 }
+      end
+
+      it 'to include id' do
+        expect(subject.to_xml_params).to match %r[<id>1</id>]
+      end
+
+      it 'to include description' do
+        expect(subject.to_xml_params).to match %r[<description>Description#1</description>]
+      end
+
+      it 'to include quantity' do
```

Comment:
No, I don't care about it. Nokogiri has the responsible to build a correct xml. :P